### PR TITLE
Contract fixes

### PR DESCRIPTION
**Pull request by mds1**
Various contract improvements that don't impact functionality
    
**GrantRegistry**
- change memory parameters to calldata where possible
- tweak revert reason strings for consistency
- add validation to getter methods

**GrantRound**
- remove unused event
- tweak formatting for consistency

**MerkleGrantRoundPayout**
- fix typo in comments
- simplify assignment in _setClaimed

**GrantRoundManager**
- add SafeMath (lost it when downgrading from 0.8.x to 0.7.6)
- move GrantDonation before funds transfer to avoid potential for mismatched event ordering via reentrancy
- validate that output token of each swap is donationToken
- ensure donationRatios mapping always has storage fully cleared
- donation validation ensures the round and manager have the same registry
- remove unnecessary >= in constructor require statement
- fix typo in comments

### CODE REVIEW

File: contracts/contracts/GrantRound.sol [hunk 1]
```diff
@@ -108,12 +102,10 @@ contract GrantRound {
    * @notice Updates the metadata pointer to a new location
    * @param _newMetaPtr A string where the updated metadata is stored
    */
-  function updateMetadataPtr(string memory _newMetaPtr) external {
+  function updateMetadataPtr(string calldata _newMetaPtr) external {
     require(msg.sender == metadataAdmin, "GrantRound: Action can be perfomed only by metadataAdmin");
-    string memory oldPtr = metaPtr;
```
Comment (thelostone-mc) [inline contracts/contracts/GrantRound.sol:113]: why did we have the oldPtr in the first place ?  Can't seem to recall

File: contracts/contracts/GrantRoundManager.sol [hunk 2]
```diff
@@ -141,12 +145,13 @@ contract GrantRoundManager is SwapRouter {
       require(_donations[i].grantId < registry.grantCount(), "GrantRoundManager: Grant does not exist in registry");
 
       // Used later to validate ratios are correctly provided
-      donationRatios[_donations[i].token] += _donations[i].ratio;
+      donationRatios[_donations[i].token] = donationRatios[_donations[i].token].add(_donations[i].ratio);
```
Comment (thelostone-mc) [inline contracts/contracts/GrantRoundManager.sol:148]: why are we doing this change ?

Comment (mds1) [reply contracts/contracts/GrantRoundManager.sol:148]: We used to get SafeMath for free with 0.8.x, but 0.7.6 needs explicit usage of SafeMath, which is what we're doing here

Comment (thelostone-mc) [reply contracts/contracts/GrantRoundManager.sol:148]: Oh derp ! Gotcha

**Review by apbendi — APPROVED** [review]
Looks great. Nice work 👍

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**